playbooks/provisioning/openstack/prerequisites.yml

```diff
@@ -0,0 +1,71 @@
+---
+- hosts: localhost
+  tasks:
+
+  # Check ansible
+  - name: Check Ansible version
+    assert:
+      that: >
+        (ansible_version.major == 2 and ansible_version.minor >= 3) or
+        (ansible_version.major > 2)
+      msg: "Ansible version must be at least 2.3"
+
+  # Check shade
+  - name: Try to import python module shade
+    command: python -c "import shade"
+    ignore_errors: yes
+    register: shade_result
+  - name: Check if shade is installed
+    assert:
+      that: 'shade_result.rc == 0'
+      msg: "Python module shade is not installed"
+
+  # Check python-dns
+  - name: Try to import python DNS module
+    command: python -c "import dns"
+    ignore_errors: yes
+    register: pythondns_result
+  - name: Check if python-dns is installed
+    assert:
+      that: 'pythondns_result.rc == 0'
+      msg: "Python module python-dns is not installed"
+
+  # Check jinja2
+  - name: Try to import jinja2 module
+    command: python -c "import jinja2"
+    ignore_errors: yes
+    register: jinja_result
+  - name: Check if jinja2 is installed
+    assert:
+      that: 'jinja_result.rc == 0'
+      msg: "Python module jinja2 is not installed"
+
+  # Check Glance image
+  - name: Try to get image facts
+    os_image_facts:
+      image: "{{ openstack_default_image_name }}"
+    register: image_result
+  - name: Check that image is available
+    assert:
+      that: "image_result.ansible_facts.openstack_image"
+      msg: "Image {{ openstack_default_image_name }} is not available"
+
+  # Check network name
+  - name: Try to get network facts
+    os_networks_facts:
+      name: "{{ openstack_external_network_name }}"
+    register: network_result
+  - name: Check that network is available
+    assert:
+      that: "network_result.ansible_facts.openstack_networks"
+      msg: "Network {{ openstack_external_network_name }} is not available"
+
+  # Check keypair
+  - name: Try to show keypair
+    command: openstack keypair show {{ openstack_ssh_public_key }}
```

Comment:
I've just realized that we don't actually use the python-openstackclient anywhere in the playbooks and since it's a bunch of complex dependencies, I'd like it to stay that way.

Turns out, it's reasonably simple to do this with `shade` (which we do depend on) directly. Just replace the command above with this one:

    python -c 'import shade; cloud = shade.openstack_cloud(); exit(cloud.get_keypair("{{ openstack_ssh_public_key }}") is None)'

Sorry to ask for another change but could you do that, please?